### ISSUE TITLE
Using Logs

### ISSUE THREAD

**Issue by AllanKlaus** [socketry/falcon#90]
I'm trying to use falcon on production in one app that I'm creating. Now I'm blind of logs, I don't know what happen when I run the app it don't show any logs of requests only the database interactions.

I'm running my app using `foreman` with the command `falcon serve -b http://0.0.0.0:9292`. Everything is running on a container. I didn't change anything on the `config/enviroment/production.rb` it's running all the rails default.

Can someone help me with this?

**Comment by ioquatix** [comment]
The simplest way to get more information is to run with `--verbose`.

That being said, it produces a lot of information.

We should probably provide a middle ground for just logging requests.

Logging like this impacts performance, so it's not by default.

A simple solution which works everywhere is to use a logging middleware.

I'll think about the best way to solve this issue.

**Comment by AllanKlaus** [comment]
i will try it. thanks. If there is something that I can do to help, just let me know.

**Comment by buhrmi** [comment]
I found that the problem with the squelched logs in Falcon lies with foreman. Switching over to hivemind (another Procfile-based process manager) fixed it.

**Comment by ioquatix** [comment]
Do you know why it made a difference?

**Comment by buhrmi** [comment]
Not really. But in the hivemind readme (https://github.com/DarthSim/hivemind) they write `Hivemind uses pty to capture process output. That fixes any problem with log clipping, delays, and TTY colors other process management tools may have.`. So apparently it's a common thing and not Falcon-specific. Wouldn't lose sleep over it 😅

**Comment by buhrmi** [comment]
And `The problem with most of those tools is that processes you want to manage start to think they are logging their output into a file, and that can lead to all sorts of problems: severe lagging, losing or breaking colored output.`

**Comment by buhrmi** [comment]
yeah just wanted to note my findings here to help close this issue, unless you don't like closing issues 🤔😅

**Comment by nogweii** [comment]
How does one turn on verbose logging with `falcon host`? Can [the middleware](https://github.com/socketry/falcon/blob/main/lib/falcon/middleware/verbose.rb) be used directly?

**Comment by ioquatix** [comment]
Iirc add `verbose true` to your service config. When I am home later today I’ll check it.

**Comment by nogweii** [comment]
Your memory was correct, adding that did the trick!
